pkg/cache/cache.go

```diff
@@ -381,6 +388,9 @@ func defaultOpts(config *rest.Config, opts Options) (Options, error) {
 		opts.Scheme = scheme.Scheme
 	}
 
+	if opts.DefaultWatchErrorHandler == nil {
```

Comment:
How about keep this field defaults to be nil, not to set the `toolscache.DefaultWatchErrorHandler`.

We can call `sharedIndexInformer.SetWatchErrorHandler()` only if it is not nil, otherwise let the internal informer/reflector use default handler.

Reply:
~~So if it is nil, set this to the `toolscache.DefaultWatchErrorHandler`?~~

I will move this but only set it when this is not nil.  The default handler you are suggesting is set when invoking the creation of the sharedindexinformer.